website_sale_custom_filter/controllers/main.py

```diff
@@ -0,0 +1,133 @@
+                    min_max_filter_vals[f_id] = {"min": f_min_val}
+                else:
+                    min_max_filter_vals[f_id]["min"] = f_min_val
+        if req.get("max_cust_filter", False):
+            for vals in req["max_cust_filter"].split(","):
+                f_id, f_max_val = vals.split("_")
+                f_max_val = int(f_max_val)
+                if not min_max_filter_vals.get(f_id, False):
+                    min_max_filter_vals[f_id] = {"max": f_max_val}
+                else:
+                    min_max_filter_vals[f_id]["max"] = f_max_val
+        return min_max_filter_vals
+
+    def get_custom_checkbox_filter_values(self):
+        req = dict(request.httprequest.args)
+        value_filter_data = {}
+        if req.get("cust_filter"):
+            filter_data = req["cust_filter"].split("&")
+            for data in filter_data:
+                f_id, val = data.split("-")
+                if f_id in value_filter_data.keys():
+                    value_filter_data[f_id].append(int(val))
+                else:
+                    value_filter_data[f_id] = [int(val)]
+        return value_filter_data
+
+    def get_custom_range_filter_values(self):
+        cust_filter_min_max_vals = self.get_min_max_custom_filters()
+        curr_website = request.env["website"].get_current_website()
+        numerical_filters_available_on_website = request.env[
+            "website.sale.custom.filter"
+        ].search(
+            [("website_ids", "=", curr_website.id), ("filter_type", "=", "numerical")]
+        )
+        range_filter_data = {}
+        # get min max for numerical filters
+        # how to handle if field from filter is related to another model?
+        for num_filter in numerical_filters_available_on_website:
+            col_name = num_filter.numerical_filter_field_id.name
+            model_name = num_filter.numerical_filter_field_id.model
+            if (
+                model_name == "product.product"
+            ):  # there are fields related to product.template sometimes
+                model_name = "product.template"
+            query = f"""
+            SELECT MIN({col_name}), MAX({col_name}) FROM {model_name.replace('.','_')}"""
+            request.env.cr.execute(query)
+            available_min_value, available_max_value = request.env.cr.fetchone()
+
+            curr_filter_id = str(num_filter.id)
+            range_filter_data[curr_filter_id] = {}
+            range_filter_data[curr_filter_id][
+                "available_min_value"
+            ] = available_min_value
+            range_filter_data[curr_filter_id][
+                "available_max_value"
+            ] = available_max_value
+            range_filter_data[curr_filter_id]["min_value"] = available_min_value
+            range_filter_data[curr_filter_id]["max_value"] = available_max_value
+            if cust_filter_min_max_vals and cust_filter_min_max_vals.get(
+                curr_filter_id
+            ):
```

Comment:
```suggestion
            if cust_filter_min_max_vals.get(curr_filter_id):
```
Nitpicking, but AFAICS `cust_filter_min_max_vals` will always be a dict, so no need to check if any key is defined in it.